tests/src/rules/export.js

```diff
@@ -510,7 +532,7 @@ context('TypeScript', function () {
           }),
           test({
             code: `
-              export function Foo();
+              export function Foo() { };
```

Comment:
why is this change needed?

Reply:
Now we will ignore overloading in order to report two errors instead of three in this case.
```
export default function a(): void;
export default function a() {}
export { x as default };
```
And this test was testing duplicate export of overloading and class.
I changed it to export of method and class to fix the test.
Also I did not find a test for method and class, the new one might be the original intention.

Reply:
hmm - i'm concerned with changing an existing test being a breaking change.

We certainly should be only reporting two here, but why are the curly braces needed?

Reply:
Without the curly braces, its node type would be `TSDeclareFunction`, which is just an overload and not a real function.
I guess this was not the original intention of this test, since there is no test for functions. We would definitely prefer to test other legal code besides export duplication.
This should not be affected in the real world, because a lone `TSDeclareFunction` will produce the error `Function implementation is missing or not immediately following the declaration.(2391)`.
The `FunctionDeclaration` that it must follow will throw our eslint error.

Reply:
let's leave the original test unchanged, and add a new test with the curly braces?

Reply:
The original test currently fails, because we no longer report errors for overriding. :)
But in the real world having overridden methods without implementations is illegal anyway, so this is probably not a breaking change.